### DISCUSSION TITLE
Various chat improvements when using "radius" mode

### DISCUSSION THREAD

**Discussion by Puremin0rez** [EssentialsX/Essentials#4509] [Ideas and Feature Suggestions] [answered]
I would love to see these improvements implemented, as it would help greatly with allowing me to switch to using EssX chat. I am **so close** to being able to make the switch. I feel like many others (both players and server owners alike) would enjoy these improvements, too. The idea is that this improves what already exists by making it more configurable, while keeping the aspect of EssX chat being simple. With EssX discord bridge now being a stable release, many more server owners may be considering using EssX chat and I feel like this is as good of time as any to help make it a more enticing option.

- Add a configuration option for the default shouting state. Currently, when `radius` is specified, local chat becomes the default method of chat and prefixing your chat message with `!` or using `/toggleeshout` is required to switch to global. I would love the ability to reverse this and make the default shout state be enabled (and thus global chat by default) and (optionally) `!` to send to local instead.
- Make shouting state persistent between restarts. Currently, shouting state (via `/toggleshout`) persists between login/logout as long as the server remains online. Since it already does this much, it makes sense to make it persist between restart as well? This would change current behavior, so maybe a configuration option if retaining this behavior is important?
- Add a configuration option for enabling / disabling question chat. Currently, when `radius` is specified, question chat (`?` prefix) becomes available even when it has nothing to do with local chat. I think it makes sense for question chat to be a separate feature from local chat? I am aware this can be semi disabled by not giving the permission, but I would prefer an *actual* explicit disable of the feature so that things like the chat prefix `?` are not parsed rather than giving a lacking perms message.
- Add a permission for sending messages to the local channel. Shout and question channels already support this, but local does not. I think it makes sense to have send permissions for all 3 chat channels, as there is currently no way to prevent a player from speaking in local.
- Add permissions for receiving messages from channels (shout, local, question). Currently, there are permissions for controlling if a player is able to send messages to these channels (minus local for some reason), but not for receiving messages from these channels. A normal use case would be having question chat be something only a helper group on a server can receive to provide support in a private manner. A more creative use case would be rebranding question chat to be a staff chat via the lang file, and giving send & receive permissions only to staff members. Potentially very flexible in a lot of ways.
- Add a space between the local chat prefix and the username by default. Currently, local chat is missing a space between the `[L]` and `<username>` - which differs from both shout and question chat (as well as normal non `radius` mode chat), which properly have a space between the channel and username. Purely cosmetic, but it'd look a lot better.
- Give some color to the local chat prefix `[L]` by default. It's currently white, and both shout and question have a nice colored prefix, so it might make sense to apply some color to local as well? Opinionated and cosmetic, just think it'd look better.

I understand the importance of retaining backwards compatibility but feel these improvements can hopefully be mostly implemented without harming such, and would lead to massive improvements in making these severely underutilized chat features be viable for many server owners as well as just making it appear more polished.

Thank you for reading and taking my ideas into consideration! **_#MakeEssXChatGreatAgain_**

**Comment by JRoy** [comment, marked as answer]
A pull request has been opened and will now track these improvements: https://github.com/EssentialsX/Essentials/pull/4535
